### PR TITLE
Convert print to logger format for much more granular unified control.

### PR DESCRIPTION
**Pull request by konantian**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [ ] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [x] I've assigned a reviewer

#### PR contents

- [x] I've consulted [ivadomed's internal developer documentation](https://github.com/ivadomed/ivadomed/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/ivadomed/ivadomed/wiki/tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/ivadomed/ivadomed/wiki/documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

This PR aims to convert all the `print` to `logger` format for much more granular unified control and  enable centralized logging to files all the much easier in the future.

### CODE REVIEW

File: testing/unit_tests/test_training_time.py [hunk 1]
```diff
@@ -199,12 +200,12 @@ def test_unet_time(download_data_testing_test_files, train_lst, target_lst, conf
...
-    print('Mean SD pred {} -- {}'.format(np.mean(pred_lst), np.std(pred_lst)))
-    print('Mean SDopt {} --  {}'.format(np.mean(opt_lst), np.std(opt_lst)))
-    print('Mean SD gen {} -- {}'.format(np.mean(gen_lst), np.std(gen_lst)))
-    print('Mean SD scheduler {} -- {}'.format(np.mean(schedule_lst), np.std(schedule_lst)))
+    logger.debug('Mean SD init {} -- {}'.format(np.mean(init_lst), np.std(init_lst)))
+    logger.debug('Mean SD load {} -- {}'.format(np.mean(load_lst), np.std(load_lst)))
+    logger.debug('Mean SD pred {} -- {}'.format(np.mean(pred_lst), np.std(pred_lst)))
+    logger.debug('Mean SDopt {} --  {}'.format(np.mean(opt_lst), np.std(opt_lst)))
+    logger.debug('Mean SD gen {} -- {}'.format(np.mean(gen_lst), np.std(gen_lst)))
+    logger.debug('Mean SD scheduler {} -- {}'.format(np.mean(schedule_lst), np.std(schedule_lst)))
```
Comment (dyt811) [inline testing/unit_tests/test_training_time.py:208]: info

File: testing/unit_tests/test_training_time.py [hunk 2]
```diff
@@ -128,7 +129,7 @@ def test_unet_time(download_data_testing_test_files, train_lst, target_lst, conf
     model_class = getattr(imed_models, model_params["name"])
     model = model_class(**model_params)
 
-    print("Training {}".format(model_params["name"]))
+    logger.debug("Training {}".format(model_params["name"]))
```
Comment (dyt811) [inline testing/unit_tests/test_training_time.py:132]: info

File: testing/unit_tests/test_transfer_learning.py [hunk 3]
```diff
@@ -21,36 +22,36 @@ def setup_function():
 @pytest.mark.parametrize('path_model', [str(Path(__data_testing_dir__, 'model_unet_test.pt'))])
 def test_transfer_learning(download_data_testing_test_files, path_model, fraction, tolerance=0.15):
     device = torch.device("cpu")
-    print("Working on {}.".format('cpu'))
-    print(__data_testing_dir__)
+    logger.debug("Working on {}.".format('cpu'))
+    logger.debug(__data_testing_dir__)
```
Comment (dyt811) [inline testing/unit_tests/test_transfer_learning.py:26]: info

File: testing/unit_tests/test_transfer_learning.py [hunk 4]
```diff
@@ -21,36 +22,36 @@ def setup_function():
...
 
     # Load pretrained model
     model_pretrained = torch.load(path_model, map_location=device)
     # Setup model for retrain
     model_to_retrain = imed_models.set_model_for_retrain(path_model, retrain_fraction=fraction,
                                                          map_location=device)
 
-    print('\nSet fraction to retrain: ' + str(fraction))
+    logger.debug(f"\nSet fraction to retrain: {fraction}")
```
Comment (dyt811) [inline testing/unit_tests/test_transfer_learning.py:34]: info

File: testing/unit_tests/test_transfer_learning.py [hunk 5]
```diff
@@ -21,36 +22,36 @@ def setup_function():
...
 
     # Check Frozen part
     grad_list = [param.requires_grad for name, param in model_to_retrain.named_parameters()]
     fraction_retrain_measured = sum(grad_list) * 1.0 / len(grad_list)
-    print('\nMeasure: retrained fraction of the model: ' + str(round(fraction_retrain_measured, 1)))
+    logger.debug(f"\nMeasure: retrained fraction of the model: {round(fraction_retrain_measured, 1)}")
```
Comment (dyt811) [inline testing/unit_tests/test_transfer_learning.py:39]: info

File: testing/unit_tests/test_transfer_learning.py [hunk 6]
```diff
@@ -21,36 +22,36 @@ def setup_function():
...
     # for name, param in model.named_parameters():
     #    print("\t", name, param.requires_grad)
     assert (abs(fraction_retrain_measured - fraction) <= tolerance)
     total_params = sum(p.numel() for p in model_to_retrain.parameters())
-    print('{:,} total parameters.'.format(total_params))
+    logger.debug('{:,} total parameters.'.format(total_params))
```
Comment (dyt811) [inline testing/unit_tests/test_transfer_learning.py:44]: info

File: testing/unit_tests/test_transfer_learning.py [hunk 7]
```diff
@@ -21,36 +22,36 @@ def setup_function():
...
     total_trainable_params = sum(
         p.numel() for p in model_to_retrain.parameters() if p.requires_grad)
-    print('{:,} parameters to retrain.'.format(total_trainable_params))
+    logger.debug('{:,} parameters to retrain.'.format(total_trainable_params))
```
Comment (dyt811) [inline testing/unit_tests/test_transfer_learning.py:47]: info

File: testing/unit_tests/test_transfer_learning.py [hunk 8]
```diff
@@ -21,36 +22,36 @@ def setup_function():
...
     assert (total_params > total_trainable_params)
 
     # Check reset weights
     reset_list = [(p1.data.ne(p2.data).sum() > 0).cpu().numpy()
                   for p1, p2 in zip(model_pretrained.parameters(), model_to_retrain.parameters())]
     reset_measured = sum(reset_list) * 1.0 / len(reset_list)
-    print('\nMeasure: reset fraction of the model: ' + str(round(reset_measured, 1)))
+    logger.debug(f"\nMeasure: reset fraction of the model: {round(reset_measured, 1)}")
```
Comment (dyt811) [inline testing/unit_tests/test_transfer_learning.py:54]: info

**Review by dyt811 — APPROVED** [review]
A few logging level changes here and there see commends. 

Biggest change you need to ensure all of them have the loguru import logger as some of them are still using default logger. 

Once those changes are made, feel free to merge.

File: dev/class_balance.py [hunk 9]
```diff
@@ -55,7 +52,7 @@ def run_main(args):
 
     balance_dct = {}
     for ds_lst, ds_name in zip([train_lst, valid_lst, test_lst], ['train', 'valid', 'test']):
-        print("\nLoading {} set.\n".format(ds_name))
+        logger.info("\nLoading {} set.\n".format(ds_name))
```
Comment (dyt811) [inline dev/class_balance.py:55]: Could you update these to f string format before merge? Thanks.

File: ivadomed/utils.py [hunk 10]
```diff
@@ -406,3 +407,9 @@ def init_ivadomed():
     """Initialize the ivadomed for typical terminal usage."""
     # Display ivadomed version
     logger.info('\nivadomed ({})\n'.format(__version__))
+
+
+def print_stats(arr):
+    logger.info(f"\tMean: {np.mean(arr)} %")
+    logger.info(f"\tMedian: {np.median(arr)} %")
+    logger.info(f"\tInter-quartile range: [{np.percentile(arr, 25)}, {np.percentile(arr, 75)}] %")
```
Comment (dyt811) [inline ivadomed/utils.py:415]: 👍 Thanks for consolidating this.

File: testing/common_testing_util.py [hunk 11]
```diff
@@ -47,17 +47,16 @@ def download_dataset(dataset: str = 'data_testing', verbose=True):
...
 
     Args:
         dataset (str): the name of the dataset to remove
-        verbose (bool): whether or not to print
     """
 
     path_dataset = Path(path_temp, dataset)
 
-    printv("rm -rf %s" % (path_dataset), verbose=verbose, type="code")
+    logger.debug("rm -rf %s" % path_dataset)
```
Comment (dyt811) [inline testing/common_testing_util.py:59]: F string update.

File: testing/unit_tests/test_training_time.py [hunk 12]
```diff
@@ -129,7 +130,7 @@ def test_unet_time(download_data_testing_test_files, train_lst, target_lst, conf
...
```
Comment (dyt811) [inline testing/unit_tests/test_training_time.py:133]: F string update.

**Review by dyt811 — COMMENTED** [review]
Good stuff. 

Minor few areas that have a few missing f string update. 

Aside from that, no issues blocking updates.

File: dev/plot_film_parameters.py [hunk 13]
```diff
@@ -101,7 +102,7 @@ def visualize_tsne(data, metadata_values, layer_no, fname_out):
 
     tsne_df = pd.concat([tsne_df, tsne_df2])
 
-    print('t-SNE done!')
+    logger.info('t-SNE done!')
```
Comment (arbalest2) [inline dev/plot_film_parameters.py:105]: Some of the print statements use a single quote for the string whereas others use double quote. You might want to use a code formatter (like [black](https://black.readthedocs.io/en/stable/)) that will unify them.
